.github/workflows/ci-slang-test.yml

```diff
@@ -0,0 +1,211 @@
+        run: |
+          PATH=$bin_dir:$PATH tools/slangc-test/test.sh
+      - name: Test Slang via glsl
+        # Run GLSL backend tests on release for pull requests, and not on merge_group, to reduce CI load.
+        if: inputs.platform != 'wasm' && inputs.os != 'macos' && inputs.full-gpu-tests && inputs.config == 'release' && github.event_name == 'pull_request'
+        run: |
+          export SLANG_RUN_SPIRV_VALIDATION=1
+          export SLANG_USE_SPV_SOURCE_LANGUAGE_UNKNOWN=1
+          "$bin_dir/slang-test" \
+            -use-test-server \
+            -server-count ${{ inputs.server-count }} \
+            -category ${{ inputs.test-category }} \
+            -emit-spirv-via-glsl \
+            -api vk \
+            -expected-failure-list tests/expected-failure-via-glsl.txt \
+            -skip-reference-image-generation \
+            -show-adapter-info
+
+  test-slang-rhi:
+    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+          fetch-depth: "2"
+
+      - name: Common Test Setup
+        uses: ./.github/actions/common-test-setup
+        with:
+          os: ${{ inputs.os }}
+          compiler: ${{ inputs.compiler }}
+          platform: ${{ inputs.platform }}
+          config: ${{ inputs.config }}
+
+      - name: Run slang-rhi tests
+        # Run slang-rhi tests on debug+release for pull requests, and only on release for merge_group, to reduce CI load.
+        # Some of the expensive tests that are not relevant for Slang (because they just test graphics API related things) are excluded using -tce.
+        if: inputs.platform != 'wasm' && inputs.full-gpu-tests && (github.event_name == 'pull_request' || inputs.config == 'release')
+        run: |
+          export SLANG_RHI_EXCLUDE_TESTS="md-clear*,cmd-copy*,cmd-upload*,fence*,staging-heap*,texture-create*"
+          if [[ "${{ inputs.os }}" == "macos" ]]; then
+            export SLANG_RHI_EXCLUDE_TESTS="sampler-array"
+          fi
+          "$bin_dir/slang-rhi-tests" -check-devices -tce="$SLANG_RHI_EXCLUDE_TESTS"
+
+  test-slangpy:
+    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+          fetch-depth: "2"
+
+      - name: Common Test Setup
+        uses: ./.github/actions/common-test-setup
+        with:
+          os: ${{ inputs.os }}
+          compiler: ${{ inputs.compiler }}
+          platform: ${{ inputs.platform }}
+          config: ${{ inputs.config }}
+
+      - name: Run slangpy tests
+        # Run slangpy tests on debug+release for pull requests, and only on release for merge_group, to reduce CI load.
+        if: inputs.platform != 'wasm' && inputs.full-gpu-tests && (github.event_name == 'pull_request' || inputs.config == 'release')
```

Comment:
A further cleanup is the refactor this `if` condition as a parameter, whose value is set from ci.yml directly, instead of using complex conditions here.